pkg/client/delegation/redelegation.go

```diff
@@ -1,16 +1,27 @@
 package delegation
 
+// TODO_TECHDEBT(@h5law): This is disgusting get this piece of shit out the codebase
```

Comment:
Haha, while I do appreciate this, any chance you can just another sentence on what you mean for others. Is it the whole file?